skipper_test.go

```
@@ -0,0 +1,98 @@
+package skipper
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/zalando/skipper/filters/builtin"
+	"github.com/zalando/skipper/proxy"
+	"github.com/zalando/skipper/routing"
+)
+
+func initializeProxy() *http.Handler {
+	filterRegistry := builtin.MakeRegistry()
+	proxy := proxy.New(routing.New(routing.Options{
+		FilterRegistry: filterRegistry,
+		DataClients:    []routing.DataClient{}}), proxy.OptionsNone)
+	return &proxy
+}
+
+func TestMissingOptionHTTP(t *testing.T) {
+	o := Options{}
+	if o.isHTTPS() {
+		t.FailNow()
+	}
+}
+
+func TestWithOptionHTTPS(t *testing.T) {
+	o := Options{CertPathTLS: "foo", KeyPathTLS: "bar"}
+	if !o.isHTTPS() {
+		t.FailNow()
+	}
+}
+
+func TestWrongCert(t *testing.T) {
+	o := Options{Address: ":9091",
+		CertPathTLS: "fixtures/notFound.crt",
+		KeyPathTLS:  "fixtures/test.key",
+	}
+	proxy := initializeProxy()
+
+	err := listenAndServe(proxy, &o)
+	if err == nil {
+		t.Fatal(err)
+	}
+}
+
+func TestWrongKey(t *testing.T) {
+	o := Options{Address: ":9091",
+		CertPathTLS: "fixtures/test.crt",
+		KeyPathTLS:  "fixtures/notFound.key",
+	}
+	proxy := initializeProxy()
+	err := listenAndServe(proxy, &o)
+	if err == nil {
+		t.Fatal(err)
+	}
+}
+
+func TestHTTPSServer(t *testing.T) {
+	o := Options{
+		Address:     ":9091",
+		CertPathTLS: "fixtures/test.crt",
+		KeyPathTLS:  "fixtures/test.key",
+	}
+	proxy := initializeProxy()
+	go listenAndServe(proxy, &o)
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	r, err := client.Get("https://localhost:9091")
+	if err == nil {
```

Right, I can probably just ignore the error then.